plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlSourceValidatorTest.java

```diff
@@ -173,4 +180,29 @@ public void testGetProject() throws CoreException {
     assertEquals(project, testProject);
   }
 
+  @Test
+  public void testNoNullPointerExceptionOnNonFacetedProject()
+      throws CoreException, ValidationException {
+    IProjectDescription projectDescription =
+        ResourcesPlugin.getWorkspace().newProjectDescription("non-faceted-project");
```

Comment:
Would it would be better to move this into the TestProjectCreator, and change it to only add the `FacetedProjectNature.NATURE_ID` if there are facets to install?

Reply:
Done. Let's see if this cause any test failure in other classes.